app.js

```diff
@@ -249,44 +277,292 @@ const app = Vue.createApp({
+                hamming_distance_AR = AR_levels[this.m("AR")]-AR_levels[this.extractValueMetric("AR",tmp_vector)]
+
+
+                //if any is less than zero this is not the right max
+                if (hamming_distance_AV<0 || hamming_distance_PR<0 || hamming_distance_UI<0 || hamming_distance_AC<0 || hamming_distance_AT<0 || hamming_distance_VC<0 || hamming_distance_VI<0 || hamming_distance_VA<0 || hamming_distance_SC<0 || hamming_distance_SI<0 || hamming_distance_SA<0 || hamming_distance_CR<0 || hamming_distance_IR<0 || hamming_distance_AR<0) {
+                    continue
+                }
+                else{
+                    //if multiple maxes exist to reach it it is enough the first one
+                    break
+                }
             }
-            else if(lookup[1] == "1") {
-                value = value + AC_diff[this.m("AC")] + AT_diff[this.m("AT")]
+
+            
+
+            current_hamming_distance_eq1 = hamming_distance_AV + hamming_distance_PR + hamming_distance_UI
+            current_hamming_distance_eq2 = hamming_distance_AC + hamming_distance_AT
+            current_hamming_distance_eq3eq6 = hamming_distance_VC + hamming_distance_VI + hamming_distance_VA + hamming_distance_CR + hamming_distance_IR + hamming_distance_AR
+            current_hamming_distance_eq4 = hamming_distance_SC + hamming_distance_SI + hamming_distance_SA
+            current_hamming_distance_eq5 = 0
+
+
+            step = 0.1
+
+            //if the next lower macro score do not exist the result is Nan
+            available_distance_eq1 = value - score_eq1_next_lower_macro
+            available_distance_eq2 = value - score_eq2_next_lower_macro
+            available_distance_eq3eq6 = value - score_eq3eq6_next_lower_macro
+            available_distance_eq4 = value - score_eq4_next_lower_macro
+            available_distance_eq5 = value - score_eq5_next_lower_macro
+
+            percent_to_next_eq1_hamming = 0
+            percent_to_next_eq2_hamming = 0
+            percent_to_next_eq3eq6_hamming = 0
+            percent_to_next_eq4_hamming = 0
+            percent_to_next_eq5_hamming = 0
+            
+
+            //some of them do not exist, we will find them by retrieving the score. If score null then do not exist
+            n_existing_lower = 0
+
+            normalized_hamming_eq1 = 0
+            normalized_hamming_eq2 = 0
+            normalized_hamming_eq3eq6 = 0
+            normalized_hamming_eq4 = 0
+            normalized_hamming_eq5 = 0
+
+            //multiply by step because distance is pure
+            maxHamming_eq1 = this.maxHammingData['eq1'][String(eq1_val)]*step
+            maxHamming_eq2 = this.maxHammingData['eq2'][String(eq2_val)]*step
+            maxHamming_eq3eq6 = this.maxHammingData['eq3'][String(eq3_val)][String(eq6_val)]*step
+            maxHamming_eq4 = this.maxHammingData['eq4'][String(eq4_val)]*step
+
+
+            if (!isNaN(available_distance_eq1)){
```

Comment:
Start of repeated/similar code blocks, needs a better data structure and a loop or similar approach.